json.go

```diff
@@ -52,6 +52,17 @@ func ErrorResponse(err error) JSONResponse {
 	return MessageResponse(500, err.Error())
 }
 
+// Error Responses in the standard Matrix Error format (errcode / error)
+func MatrixErrorResponse(httpStatusCode int, errCode, message string) *JSONResponse {
```

Comment:
Could you change this to return a `JSONResponse` instead to be consistent with the other methods?